async-stream-impl/Cargo.toml

```diff
@@ -18,7 +18,7 @@ syn = { version = "1", features = ["extra-traits", "full", "visit-mut"]}
 quote = "1"
 
 [dev-dependencies]
-async-stream = { version = "0.3.0", path = "../async-stream" }
```

Comment:
We can avoid the circular dev-dependencies problems by removing the version from path dev-dependencies. See https://github.com/rust-lang/futures-rs/pull/2305 for more.